### ISSUE TITLE
Reveal in Mobile Safari is broken for iPads

### ISSUE THREAD

**Issue by ingueferroque** [zurb/foundation-sites#10487]
#### How to reproduce this bug:

1. Go to http://foundation.zurb.com/sites/docs/reveal.html in Safari on iPad or desktop
2. When on desktop Safari, go to Develop - User Agent and switch to 'Safari - IOS 10 - iPad'
3. On the Reveal docs page, click on the 'Click me for a modal' button

#### What should happen:

The background should be rendered correctly, as it would normally be shown but with a dark overlay.

#### What happened instead:

On large and wider screens a big white band is being rendered to the right of the screen, the background site is not being rendered beyond the large breakpoint.

#### Browser(s) and Device(s) tested on:

Safari 10.1.2 on Mac OS 10.12.6
Latest version of Safari on iPad Pro

#### Foundation Version(s) you are using:

6.4.1

#### Test case link:

http://foundation.zurb.com/sites/docs/reveal.html

**Comment by circlingthesun** [comment]
This happens on iPhone also. It appears to be related to the 'is-reveal-open' class being added to the html element.

**Comment by rafibomb** [comment]
I'm not seeing this on the docs, though I had this issue on my last project where the only thing on the page was the button that opened it. I ended up solving it by adding a wrapper inside the body that was 100vw wide.

I tested it with setting 
```
.is-reveal-open {
  overflow: visible !important;
}
```

and it solves this. Now I'm wondering why this property was used and what consequences will result by changing it.

**Comment by rafibomb** [comment]
I hesitate to change this since every modal I've seen does this. I think BS example:
```
.modal-open {
  overflow: hidden;
}
```

Still looking for a better solution

**Comment by circlingthesun** [comment]
@rafibomb on Chrome I notice two scrollbars showing up when overflow is hidden. On IOS the is-reveal-open class is set on both the html and body element. The issue appears to be solved by adding:

```
html.is-reveal-open {
    overflow: inherit;
}

body.is-reveal-open {
    overflow: hidden;
}
```

You can figure out the rationale behind the code by toggling the blame view in the source: https://github.com/zurb/foundation-sites/blame/develop/scss/components/_reveal.scss#L136

**Comment by circlingthesun** [comment]
So: https://github.com/zurb/foundation-sites/commit/2087bedf6c7afac7fba3a28ae8350ebfa10bdb5a

**Comment by ingueferroque** [comment]
This is still an issue for me in 10.4.2, unfortunately. But the tweaks posted here do help make it functional.  Thank you all for your help looking into this!

**Comment by jimmy72** [comment]
I had the same issues with the reveal on my ipad. I had been searching for 2 days to find a solution for this untill I found this open issue. 
It seems that the suggested solution of circlingthesun works indeed. Thank you! You made my day!


html.is-reveal-open {
    overflow: inherit;
}

body.is-reveal-open {
    overflow: hidden;
}

But can anyone tell me why it looks as if the page underneath the reveal (on opening it) is being squeezed to a smaller width on phones and tablets?

Thanks 

Jimmy

**Comment by zxtom** [comment]
Just tested this on an iPad running the latest Safari (iOS11, iPad Pro).

1. No issues with the white band as reported by ingueferroque.
2. What I've noticed is that revealing a modal (by tapping on any of the 'click me' buttons on http://foundation.zurb.com/sites/docs/reveal.html) forces the browser to scroll to the top of the page and reveal the address bar; once the modal is closed the page position returns to where it was before tapping on the button (but the address bar stays visible). Also, as reported by jimmy72, some content in the page underneath seem to be changing position / size.

This happens both in portrait and landscape orientation. I can try taking some screenshots if that helps.

**Comment by Proper-Job** [comment]
For what it's worth, I was able to mitigate the problem by adding:
```css
body {
    width: 100%
}
```
My use case allows for this rule.

**Comment by DanielRuf** [comment]
I guess this is a bug in Safari 10.

**Comment by ncoden** [comment]
Not a Safari bug. With `.is-reveal-open`, `<body>` get `position: fixed` so it is put out of the flow and doesn't act like a normal block like taking the full width. Instead, it takes the width of its content. It's hard to say why this bug doesn't appear on others browsers, but the error is on our side anyway. We should apply `min-width: 100%;`.

**Comment by ncoden** [comment]
Already resolved in https://github.com/zurb/foundation-sites/pull/10583. WIll be released in `v6.5.0`
